forward/events.go

```diff
@@ -0,0 +1,64 @@
+package forward
+
+// An EventListener handles events given to it by the SWIM node. HandleEvent should be thread safe.
+type EventListener interface {
+	HandleEvent(interface{})
```

Comment:
I think we're accruing debt by adding to the public/exported interfaces of our types. It's an API design smell when parts of an API are meant for only certain callers. I think we should not perpetuate bad patterns.